libsolidity/formal/CHC.cpp

```diff
@@ -583,7 +583,15 @@ void CHC::externalFunctionCall(FunctionCall const& _funCall)
 	/// so we just add the nondet_interface predicate.
 
 	solAssert(m_currentContract, "");
-
+	if (MemberAccess const* fun = dynamic_cast<MemberAccess const*>(&_funCall.expression()))
```

Comment:
If `isTrustedCode` takes an `Expression` instead, here you could just test `isTrustedCode(_funCall.expression())`, and function `isTrustedCode` could do the unwrapping of the Identifier from the MemberAccess, if it is a MemberAccess

Reply:
Changed according to the suggestion.